libdrgn/dwarf_info_cache.c

```diff
@@ -628,6 +628,7 @@ drgn_compound_type_from_dwarf(struct drgn_dwarf_info_cache *dicache,
 		*ret = type;
 		return NULL;
 	}
+	drgn_update_language_from_dwarf(type, die);
```

Comment:
Even though it will cause churn, I think it'd be cleaner to pass the language to all of the initializers. This way, we're explicit about the places that we're setting a default language for now. Plus, drgn_types should be immutable, and having a separate setter for the language makes that less obvious.

So, let's add `const struct drgn_language *lang` as the last argument to all of the initializers, make use of it in the DWARF code, and explicitly pass in `&drgn_language_c` where we don't yet have that information.